tests/imap/test_save_folder_names.py

```diff
@@ -19,7 +19,7 @@ def test_imap_save_generic_folder_names(db, default_account):
     raw_folders = [RawFolder(*args) for args in folder_names_and_roles]
     monitor.save_folder_names(db.session, raw_folders)
     saved_folder_data = set(
-        db.session.query(Folder.name, Folder.canonical_name).filter(
+        db.session.query(Folder.name, Folder._canonical_name).filter(
```

Comment:
I think the main problem is the `@property` decorator actually returns a `property object` at runtime. SQLAlchemy checks the type of the object eagerly and throws an error if it's not something it expects. If I just use `canonical_name` instead of `_canonical_name`, I get the following error:

```
E               InvalidRequestError: SQL expression, column, or mapped entity expected - got '<property object at 0x3cf4158>'
```